documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -1013,3 +1027,228 @@ Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rul
+
+**Why the rule is important**: [ARM guidelines](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md) enforce this rule. 
+
+**How to fix the violation**: Consider moving extra properties into "properties" bag of the resource model.
+
+**Examples**: 
+**Bad example**: "extraProperty" is not allowed at top level of the resource model.
+```json
+"VersionedApplicationType": {
+  "description": "The versioned application type resource",
+  "properties": {
+    "extraProperty": {
+      "type": "string",
+      "description": "Extra property description"
+    }
+  },
+  "allOf": [
+  {
+    "$ref": "#/definitions/Resource"
+  }
+  ]
+}
+```
+**Good example**: Notice that "extraProperty" is inside "properties" bag, and not at top level.
+```json
+"VersionedApplicationType": {
+  "description": "The versioned application type resource",
+  "properties": {
+    "properties":{
+      "extraProperty": {
+        "type": "string",
+        "description": "Extra property description"
+      }
+    }
+  },
+  "allOf": [
+  {
+    "$ref": "#/definitions/Resource"
+  }
+  ]
+}
+```
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2057" />R2057 InvalidSkuModel
+**Output Message**: Sku Model definition '{0}' is not valid. A Sku model must have 'name' property. It can also have 'tier', 'size', 'family', 'capacity' as optional properties.
+
+**Description**: Sku model definition needs to follow [ARM guidelines](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md) and can contain only a certain set of properties as described in the message. 
+
+**Why the rule is important**: Per [ARM guidelines](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md) this rule is enforced.
+
+**How to fix the violation**: Update the sku model definition.
+
+**Examples**: N/A
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2018" />R2018 XmsEnumValidation
+**Output Message**: The enum types should have x-ms-enum type extension set with appropriate options. Property name: {0}.
+
+**Description**: AutoRest defines [x-ms-enum extension](https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-enum) to provide more flexibily for enum types, please refer to the documentation.
+
+**Why the rule is important**: Including [x-ms-enum extension](https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-enum) provides more flexibilty for enum types in SDK generated code.
```

Comment:
nit: As a side note, I feel we should talk a little about using modelAsString: true as a default here.